### DISCUSSION TITLE
Generic in TRCPClientError

### DISCUSSION THREAD

**Discussion by kalempster** [trpc/trpc#3920] [Q&A]
I have a mutation that I want to catch if the procedure throws an error, I'd like to get the code of the error ("UNAUTHORIZED", "FORBIDDEN" etc.) but there are no types on TRCPClientError with the property code. I saw a generic that you can pass to the error but I don't know what to pass as the generic to get intelisense.
```ts
try {
    await mutateAsync(id);
} catch (error) {
    if(error instanceof TRPCClientError<>){
        //Process the error here if unauthorized regenerate accessToken and retry
    }
}
```
Is there any way to do so?

**Comment by juliusmarminge** [comment]
You can do this via the `errorFormatter` when you initialize tRPC:

```ts
const t = initTRPC.create({
  errorFormatter: ({ shape, error }) => {
    return {
      ...shape,
      code: error.code, // UNAUTHORIZED | INTERNAL_SERVER_ERROR | BAD_REQUEST ...
    };
  },
});
```

Then you'll have it available on the client:

```ts
export default function IndexPage() {
  const result = trpc.greeting.useQuery({ name: 'client' });

  const error = result.error;
  error?.shape?.errorCode;

  return ( ... );
}
```
